### DISCUSSION TITLE
Use "if-else" in input file & set initial value for post-processor

### DISCUSSION THREAD

**Discussion by hokonkim** [idaholab/moose#18247] [Q&A General] [answered]
Because I'm working on several themes simultaneously, I made several discussion...

I'm sorry about that.

First question is

**"How to use "if-else" statement in an input file?"**

<img width="570" alt="Screen Shot 2021-07-02 at 6 41 46 PM" src="https://user-images.githubusercontent.com/46009457/124334584-86856100-db65-11eb-9c9f-ae1ef9f246a1.png">

I tried to use "if-else" statement, but this failed.. I also tried to find an appropriate example for what I need. It was hard to find.

for example, what I want to do is 

if (time=0) {'vals = 0 1.03665 0.4'}
else {'vals = 0.03607 1.03665 0.4'}

Second question is

**"How to assign the initial value (initialization) for post-processor?"**

From the previous discussion that I made, I knew Postprocessor is not being executed at the 0th timestep.

So, introducing execute_on = 'initial timestep_end' in the postprocessor will work.

However, what I want to do now is "assign a specific value for postprocessor at 0th step"

For my function like below,

<img width="1168" alt="Screen Shot 2021-07-02 at 6 49 02 PM" src="https://user-images.githubusercontent.com/46009457/124334790-38249200-db66-11eb-85da-7c152ab88512.png">

I need to assign 0 on postprocessor "numerical_phi_YSZ" at 0th step to make it work correctly.

**Comment by GiudGiud** [comment, marked as answer]
Hello
No this is fine, a discussion for each topic is definitely the way to go. 

In general, you can use the Controls system to enable/disable blocks, to make if-else like statements. 
But for a parsed function, you can use this syntax:
```
  [./const_ref]
    type = ParsedFunction
    value = '
            ix := if(x < 0.5, 0, if(x < 1, 1, 2));
            iy := if(y > 0, 2, if(y > -0.5, 1, 0));
            iy * 3 + ix
            '
  [../]
```

You cant do that for postprocessors. If executing at the initial timestep does not work for you, then you need to use an if-else statement there too. 

Guillaume

**Reply by GiudGiud** [reply]
First you dont need this:
material_property_names = k
k does not come in the `function` expression

then I just tested your block with this input file below and it gave 0.024 as expected:
```
[Mesh]
  type = GeneratedMesh
  dim = 2
  nx = 10
  ny = 10
[]

[Variables]
  [u]
  []
[]

[Kernels]
  [diff]
    type = Diffusion
    variable = u
  []
[]

[BCs]
  [left]
    type = DirichletBC
    variable = u
    boundary = left
    value = 0
  []
  [right]
    type = DirichletBC
    variable = u
    boundary = right
    value = 1
  []
[]

[Executioner]
  type = Steady
  solve_type = 'PJFNK'
  petsc_options_iname = '-pc_type -pc_hypre_type'
  petsc_options_value = 'hypre boomeramg'
[]

[AuxVariables]
  [temperature]
    type = MooseVariableFVReal
    initial_condition = 2000
  []
  [k]
    type = MooseVariableFVReal
  []
[]

[AuxKernels]
  [to_k]
    type = MaterialRealAux
    property = k
    variable = k
  []
[]

[Materials]
  [thermal_conductivity]
    type = DerivativeParsedMaterial
    function ='if(temperature<=1268,0.015 ,if(1268<temperature & temperature<=1923, 0.012, 0.024))'
    args = 'temperature'
    f_name = k
  []
[]

[Outputs]
  exodus = true
[]
```

Also please create your own post, dont hijack someone else's
